### PR TITLE
Enemy follows path

### LINKED ISSUE #28: Add automated enemy movement
We need the cats to follow a predefined path on the map

### PR DESCRIPTION
**Pull request by henri-climber**
Here's an idea on how to make the cats follow the path on the map. This is just an idea up for discussion. Since i never did sth like that in arcade it may be totally wrong. Also this is not a final code version, we would need to move a few things to constants etc.

The idea is the following and comes from this https://api.arcade.academy/en/latest/examples/follow_path.html#follow-path.

1. calculate the path that the cats should follow, a few general things
   - our path is defined by a start block with a certain id and an end block with a certain id
   - right now only paths with one possible way work (would be easy to implement tho)
2. let each cat follow the path by calculating where it needs to go next

This could close #28

### CODE REVIEW

File: src/tower/game_view.py [hunk 1]
```diff
@@ -50,3 +62,82 @@ def on_draw(self):
...
+        # contains all coordinates in the form (x, y), that the enemies should follow
+        position_list = []
+
+        # this is a 1D representation of our map containing sprites
+        all_sprites = self.scene.get_sprite_list("Tile Layer 1")
+
+        # this is a 2D representation of our map containing only the id of the sprite
+        tile_map = self.tile_map.get_tilemap_layer("Tile Layer 1").data
+
+        # get the postion of the start block
```
Comment (FieryIceStickie) [inline src/tower/game_view.py:86]: position typo

File: src/tower/sprites.py [hunk 2]
```diff
@@ -33,18 +34,71 @@ def proj_speed(self, level: int) -> float:
...
 
+    def update(self):
+        """calculates new position of enemy but only when it is alive"""
+        if not self.is_alive:
+            return
+        # current position
+        start_x = self.center_x
+        start_y = self.center_y
+
+        # get next destination postion
```
Comment (FieryIceStickie) [inline src/tower/sprites.py:73]: position typo

**Review by FieryIceStickie — DISMISSED** [review]

File: src/tower/game_view.py [hunk 3]
```diff
@@ -50,3 +62,82 @@ def on_draw(self):
...
+                if col in start_sprites:
+                    return row_index, col_index
+        return None, None
+
+    def get_next_path_position(self, row, col, tile_map, visited) -> int:
+        """Returns the next path position
+        assumes that there is only one possible path
+        """
+        # has all id's of paths
+        path_blocks = [3, 6, 5]
```
Comment (henri-climber) [inline src/tower/game_view.py:122]: should we move this to constants to keep track which of our blocks are paths?

File: src/tower/game_view.py [hunk 4]
```diff
@@ -50,3 +62,82 @@ def on_draw(self):
...
+            # get next path position
+            row, col = self.get_next_path_position(row, col, tile_map, visited)
+
+        sprite = all_sprites[row * len(tile_map[0]) + col]
+        position_list.append((sprite.center_x, sprite.center_y))
+        return position_list
+
+    def _find_start_index(self, tile_map):
+        """returns the index of the tile that represents the start of the path"""
+        start_sprites = [5]
```
Comment (henri-climber) [inline src/tower/game_view.py:110]: maybe this could be in constants as well

File: src/tower/game_view.py (hunk 3 above)
Comment (FieryIceStickie) [reply src/tower/game_view.py:122]: yeah that's a good idea

Comment (FieryIceStickie) [reply src/tower/game_view.py:122]: make it a set as well

**Review by FieryIceStickie — APPROVED** [review]
LGTM

**Pull request merged**